x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/KnnVectorQueryBuilder.java

```diff
@@ -28,18 +35,25 @@
     private final String fieldName;
     private final float[] queryVector;
     private final int numCands;
+    private List<QueryBuilder> filterQueries;
 
     public KnnVectorQueryBuilder(String fieldName, float[] queryVector, int numCands) {
         this.fieldName = fieldName;
         this.queryVector = queryVector;
         this.numCands = numCands;
+        this.filterQueries = List.of();
```

Comment:
We actually allow it to be set in the `filterQueries(...)` method. Since this builder is only meant to be used internally to Elasticsearch, I just chose the setter methods that made our code the cleanest.

Reply:
Yes, but it is a public class, and `filterQueries` are public methods (can't this class be used by Java HLRC client?). If we don't want to make `filterQueries`  final, may be just ensure that are not `null` when we set them?

Reply:
I don't think this can be used from the Java HLRC. Since it lives in an xpack module, it's not available by default -- for example, for pinned queries we had to add an extra query builder just for the HLRC (https://github.com/elastic/elasticsearch/pull/45779). 

I ended up reworking this to just add the filters instead of replacing them. I also added a note to the query builder warning that it's an internal class. Thanks for the ideas.